### PR TITLE
Electrs 0.9.0

### PR DESCRIPTION
**Pull request by erikarvstedt**
This branch is currently based on `misc-5` (https://github.com/fort-nix/nix-bitcoin/pull/402).
Relevant commits start at `update nixpkgs (electrs: 0.8.11 -> 0.9.0)`.

### CODE REVIEW

**Comment by jonasnick** [comment]
looks good so far

File: modules/versioning.nix [hunk 1]
```diff
@@ -138,6 +138,37 @@ let
...
+        - Otherwise:
+          1. Deploy the new system config to your node
+          2. Check that electrs works as expected and delete the old database:
+             rm -r '${dbPath}'
+
+        [1] https://github.com/romanz/electrs/blob/557911e3baf9a000f883a6f619f0518945a7678d/doc/usage.md#upgrading
+        [2] This is based on the bitcoin blockchain size as of 2021-09.
+            The general formula is, approximately, size_of(${dbPath}) * 0.6
+            This includes the final database size (0.4) plus some extra storage (0.2).
+      '';
```
Comment (Kixunil) [inline modules/versioning.nix:170]: Beautiful explanation!

**Comment by nixbitcoin** [comment]
Running into same problem with bitcoin whitelist here

**Comment by erikarvstedt** [comment]
Which problem?

**Comment by nixbitcoin** [comment]
Hmmm, still running into

```
Oct 06 13:36:48 ephemeralelectrs electrs[125806]: Error: bitcoin p2p failed to connect: 169.254.1.12:8333
Oct 06 13:36:48 ephemeralelectrs electrs[125806]: Caused by:
Oct 06 13:36:48 ephemeralelectrs electrs[125806]:     Connection refused (os error 111)
```
after setting
```
services.bitcoind.extraConfig = ''
  whitelist=169.254.1.16
'';
```

**Comment by erikarvstedt** [comment]
The netns test passes. Can you share your config?

**Comment by nixbitcoin** [comment]
`ip netns exec nb-bitcoind ss -tlpn` shows

```
State         Recv-Q        Send-Q               Local Address:Port               Peer Address:Port       Process
LISTEN        0             0                     169.254.1.12:8332                    0.0.0.0:*           users:(("bitcoind",pid=121375,fd=10))
LISTEN        0             0                     169.254.1.12:8334                    0.0.0.0:*           users:(("bitcoind",pid=121375,fd=23))
```

???

**Review by nixbitcoin — APPROVED** [review]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 8938eadf0c4d4995381e7eb79188a7fea8c87471
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEiDOlIkszv4ojhh+OtgROy6La5dAFAmFdu/kACgkQtgROy6La
5dDO5xAAkVZo29yquDsA722d4L3FC5XvxgaAsMeDgx250VLGz9qRtt4zwngusQB7
Bu7zIeF5IFb3RgTOdhQRTDY7iwoQkvQoVAOctGG+BYbXsupUc87g0kK0Cx3JzkUv
NNYySVnRqH0CEtLDDd4nJpwTSo8YNf6l9rMjVPgjCy2i7XwW1XsyWNivAgdnSJiS
amYc6sBjss/t2VIgjHk4qh8dLikzKyGwAtBg/waemJILBs2VR4dJnzwexg+rmilf
5BLOo3GkRckoNtj1GPR6MecXscDWQkFWcaQ/dAz3Qa/IskMP4bUljYlS19ksUhuz
dUlhnMZ/NCMKREcqo864hBGyYqF9Zc+8BSY1TwcvDnNf5KAB/He0a772fER6fvdp
EnFXSqkNIHD4RzhDKatwAmJzfhCjtORK24erqaNo/jqIIXuVUn022Sz9QLD28s2F
5ChKpkvs/nPt8nSkR30MEZhKbKF0Hod1YGmjayM9n/ayLVXnOd4sYmGVBfiFxgNQ
gB94pfdr0nTav01VHLKGSpg/8/nVHrNk0dqUkgiOJsvO4yjzH2B0xP2My5mspvwp
8yKRIdhcNtCi7bS8NJT2j6PrJE6Ck7qymd+tRwB+bOUqkg6icHbLuQk1P/sx0zhT
jB6lIG43ptuPq/73OiyAnWUeBMWChAOFopyqNgzWXuiMI1DATkY=
=HZdV
-----END PGP SIGNATURE-----
```

**Review by jonasnick — APPROVED** [review]
ACK 8938eadf0c4d4995381e7eb79188a7fea8c87471

**Pull request merged**